app/publish/publish.go

```diff
@@ -211,15 +211,23 @@ func (h Handler) saveFile(r *http.Request, userID int) (*os.File, error) {
 }
 
 // createFile opens an empty file for writing inside the account's designated folder.
-// The final file path looks like `/upload_path/{user_id}/filename_{random}.ext`,
+// The final file path looks like `/upload_path/{user_id}/{random}/filename.ext`,
 // where `user_id` is user's ID and `random` is a random string generated by ioutil.
 func (h Handler) createFile(userID int, origFilename string) (*os.File, error) {
-	path := path.Join(h.UploadPath, fmt.Sprintf("%d", userID))
-	err := os.MkdirAll(path, os.ModePerm)
+	uploadPath := path.Join(h.UploadPath, fmt.Sprintf("%d", userID))
+	err := os.MkdirAll(uploadPath, os.ModePerm)
 	if err != nil {
 		return nil, err
 	}
-	return ioutil.TempFile(path, fmt.Sprintf("%s_*", origFilename))
+	randomDir, err := ioutil.TempDir(uploadPath, "")
+	if err != nil {
+		return nil, err
+	}
+	finalFilePath := path.Join(randomDir, origFilename)
+	if err := ioutil.WriteFile(finalFilePath, []byte(origFilename), os.ModePerm); err != nil {
```

Comment:
This will just create a file containing it's own filename, why would we want that?

Reply:
I changed my key. I didn't notice that I added them, thanks for the catch.

Currently path output would look like  /upload_path/{user_id}/{random}/filename.ext`

I will change it to just be: /upload_path/{user_id}/{random}

I'll make an additional commit to add a cleanup.

Since I've been doing smaller projects I'm not as experienced in testing as I should be. I did a bit of digging after reading this and am indeed able to run the individual tests on my machine.

Reply:
We still need the original filename to be preserved. My comment was specifically about the `ioutil.WriteFile(finalFilePath, []byte(origFilename)` call, this will create a new file named after the original one and then write its own filename as a string into it. Why not just open a file and leave it empty?

Reply:
I think this is due to my own misunderstanding of the test that is being run on it, as the tests tell me that  there's no such file or directory if I don't write anything to the file and then try to open it.

Error:      	Received unexpected error:
            	            	open /tmp/20404/183730872/1Mb.dat: no such file or directory
            	Test:       	Test_fetchFile/http://127.0.0.1:38039/1Mb.dat

However if I DO write anything, it recognizes the file is there, but tells me its an invalid description.

Due to my lack of knowledge here I'm unable to determine if that's because the code I'm writing is incorrect, or if I'm unable to change the tests to pass the valid code that I wrote.

Reply:
> as the tests tell me that there's no such file or directory if I don't write anything to the file and then try to open it

Not easy to tell without seeing how you were trying to create that file.

Bad file descriptor error is due to the file being opened in read-only mode.